### PR TITLE
Fix build to handle git tags that are prefixed with a "v"

### LINKED ISSUE #272: Use of APPVEYOR_REPO_TAG_NAME doesn't handle v prefix
We've named our git tags in the format `v1.2.3` for quite a while, and is the convention git's documentation, git, linux and many others follow to allow you to easily distingish between `v1.0.0` as a tag and `1.0` which might be your 1.0 series branch off 1.0.0.

I'd like to continue doing this, however the new build set up doesn't permit this. `APPVEYOR_REPO_TAG_NAME` is used 5 times in appveyor.yml and once in common.props.

I'm looking into how to change this now, but feel free to jump in if you've got any suggestions.

/cc @fir3pho3nixx

### PR DESCRIPTION
**Pull request by jonorossi**
Fixes #272.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ init:
 - ps: >-
     if ($env:APPVEYOR_REPO_TAG -eq "true")
     {
-        Update-AppveyorBuild -Version "$env:APPVEYOR_REPO_TAG_NAME"
+        Update-AppveyorBuild -Version ($env:APPVEYOR_REPO_TAG_NAME).TrimStart("v")
```
Comment (fir3pho3nixx) [inline appveyor.yml]: This is exactly what I was thinking 👍

File: appveyor.yml [hunk 2]
```diff
@@ -35,10 +35,10 @@ on_success:
 - ps: >-
     if ($env:APPVEYOR_REPO_TAG -eq "true")
     {
-        nuget push ".\build\Castle.Core.${env:APPVEYOR_REPO_TAG_NAME}.nupkg" -ApiKey $env:NUGET_API_KEY -Source https://api.nuget.org/v3/index.json
-        nuget push ".\build\Castle.Core-log4net.${env:APPVEYOR_REPO_TAG_NAME}.nupkg" -ApiKey $env:NUGET_API_KEY -Source https://api.nuget.org/v3/index.json
-        nuget push ".\build\Castle.Core-NLog.${env:APPVEYOR_REPO_TAG_NAME}.nupkg" -ApiKey $env:NUGET_API_KEY -Source https://api.nuget.org/v3/index.json
-        nuget push ".\build\Castle.Core-Serilog.${env:APPVEYOR_REPO_TAG_NAME}.nupkg" -ApiKey $env:NUGET_API_KEY -Source https://api.nuget.org/v3/index.json
+        nuget push ".\build\Castle.Core.${env:APPVEYOR_BUILD_VERSION}.nupkg" -ApiKey $env:NUGET_API_KEY -Source https://api.nuget.org/v3/index.json
```
Comment (fir3pho3nixx) [inline appveyor.yml]: Let's watch the build output in appveyor to see if this gets set properly.

**Comment by fir3pho3nixx** [comment]
Should we try merging it?

**Comment by jonorossi** [comment]
Merged, thanks for looking over it.

**Pull request merged**